src/NLog/Targets/DatabaseTarget.cs

```diff
@@ -512,6 +515,36 @@ private void WriteEventToDatabase(LogEventInfo logEvent)
             }
         }
 
+        /// <summary>
+        /// Allows the database type to be customized if we
+        /// are not able to properly infer the database type.
+        /// </summary>
+        /// <param name="dbDataParameter">The parameter.</param>
+        /// <param name="parameterInfo"></param>
+        internal void CustomizeParameterType(IDbDataParameter dbDataParameter, DatabaseParameterInfo parameterInfo)
+        {
+            if (string.IsNullOrEmpty(parameterInfo.DbType) ||
+                string.IsNullOrEmpty(parameterInfo.DbTypePropertyName))
+                return;
+            var fullEnumNamespace = parameterInfo.DbType.Split('.');
+            var justEnumName = string.Join(".", fullEnumNamespace.Take(fullEnumNamespace.Length - 1).ToArray());
+            var justEnumValue = fullEnumNamespace.Last();
```

Comment:
are we trying to get the value of the after the last `.` ? e.g. x.y.z.GETTHISVALUE ? 

We can write this less complicated and more performant, use get `.LastIndexOf `